core/textile/bucket/bucket.go

```diff
@@ -21,30 +33,31 @@ type BucketsClient interface {
 // NOTE: all write operations should use the lock for the bucket to keep consistency
 // TODO: Maybe read operations dont need a lock, needs testing
 // struct for implementing bucket interface
-type bucket struct {
+type Bucket struct {
 	lock          sync.RWMutex
 	root          *bucketsproto.Root
-	client        Client
+	ctx           context.Context
```

Comment:
It is usually an anti-pattern to store a context in a struct.

The context should be only valid for a particular period of operation, that is why it is usually passed into running functions to control their context values and trigger them to be done.

Structs objects could exists for a long time beyond the lifetime of a context and therefore all operations on that Bucket would stop executing.

Reply:
@perfectmak Added context back in bfc5cf6.

Please check it out and let me know if there's any other issues.